### PR TITLE
[SASS-3226] Create Double taxation agreement details page

### PR DESCRIPTION
**Pull request by srinivaspach**
# Checklist

Jira subtask: SASS-____

## Developer: Before Asking Others To Review Your Code

Have you ...

- [x]  Had a QA verify that all acceptance criteria have been met, with comment(s) in the ticket? 
- [ ]  Hosted a group review of your implementation?
- [ ]  Rebased against `main`? 
- [ ]  Ensured that your commit(s) contain the ticket key?
- [ ]  Ensured that your branch is named after the ticket key?
- [ ]  Successfully run the unit tests?
- [ ]  Successfully run the integration tests?
- [ ]  Successfully run the journey tests? If not, please add a comment explaining why it wasn't necessary.
- [ ]  Ensured that the PR job has passed.

## Reviewer: Before Reviewing

Have you ...

- [ ]  Pulled the branch down?
- [ ]  Assigned yourself to this PR?
- [ ]  Moved the Jira *subtask* to “in review”?

## Developer: After Approval

Have you ...

- [ ]  Moved the Jira *subtask* to “in pipeline”?

## Developer: After Merging

Have you ...

- [ ]  Scheduled a demo? If not, please add a comment.

### CODE REVIEW

File: app/views/pensions/paymentsIntoOverseasPensions/DoubleTaxationAgreementView.scala.html [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                labelHidden=false,
+                currency=true
+            )
+            @button()
+        }
+    }
+
+@{
+    // $COVERAGE-OFF$
+}
```
Comment (edem-nsefik-hmrc) [inline app/views/pensions/paymentsIntoOverseasPensions/DoubleTaxationAgreementView.scala.html:109]: Add a newline here

File: it/controllers/pensions/paymentsIntoOverseasPensions/DoubleTaxationAgreementControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        insertCyaData(pensionsUsersData(aPensionsCYAModel), aUserRequest)
+        userDataStub(anIncomeTaxUserData.copy(pensions = Some(anAllPensionsData)), nino, taxYearEOY)
+        urlGet(fullUrl(doubleTaxAgrtUrl(taxYearEOY)), follow = false,
+          headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY, validTaxYearList)))
+      }
+
+      result.status shouldBe OK
+    }
+
+    "show page when wrong tax page is added " in {
```
Comment (edem-nsefik-hmrc) [inline it/controllers/pensions/paymentsIntoOverseasPensions/DoubleTaxationAgreementControllerISpec.scala:79]: "What does wrong tax page" mean

File: it/utils/PageUrls.scala [hunk 3]
```diff
@@ -217,6 +217,10 @@ object PageUrls extends IntegrationTest {
 
     def pensionReliefSchemeDetailsUrl(taxYear: Int, index: Int): String =
       s"$appUrl/$taxYear/overseas-pensions/payments-into-overseas-pensions/pensions-overseas-details-summary?reliefIndex=$index"
+
+    def doubleTaxationAgreementUrl: Int => Int => String =
```
Comment (edem-nsefik-hmrc) [inline it/utils/PageUrls.scala:221]: Make this a val

**Review by edem-nsefik-hmrc — CHANGES_REQUESTED** [review]

File: it/controllers/pensions/paymentsIntoOverseasPensions/DoubleTaxationAgreementControllerISpec.scala (hunk 2 above)
Comment (srinivaspach) [reply it/controllers/pensions/paymentsIntoOverseasPensions/DoubleTaxationAgreementControllerISpec.scala]: Meant to say "wrong tax year"

File: it/utils/PageUrls.scala (hunk 3 above)
Comment (srinivaspach) [reply it/utils/PageUrls.scala]: Done

File: app/views/pensions/paymentsIntoOverseasPensions/DoubleTaxationAgreementView.scala.html (hunk 1 above)
Comment (srinivaspach) [reply app/views/pensions/paymentsIntoOverseasPensions/DoubleTaxationAgreementView.scala.html]: Fixed

**Review by edem-nsefik-hmrc — APPROVED** [review]

**Pull request merged**